### DISCUSSION TITLE
Stripping Stdlib for usage in micro controllers

### DISCUSSION THREAD

**Discussion by mdrokz** [grain-lang/grain#1427] [Ideas]
Hi im trying to experiment a bit with grain and wasm, and i found out that you can run wasm3 on micro controllers but the binary payload must be <= 64kb and so i thought to strip some of the unnecessary modules from the grain stdlib and try compiling again. but it didnt work, does grain use tree shaking & dead code elimination on compilation ? and is there anyway i can achieve what im trying to do ?

**Comment by ospencer** [comment]
Hi there! There's a number of optimizations coming in Grain v0.6 (November-ish time frame) that reduce module size dramatically. I've been playing around with the Wasm4 game engine and it has a similar module size requirement.

Grain does do tree shaking and dead code elimination, but the upcoming optimizations allow them to work far more effectively.

Hope that's helpful!

**Reply by mdrokz** [reply]
> Hi there! There's a number of optimizations coming in Grain v0.6 (November-ish time frame) that reduce module size dramatically. I've been playing around with the Wasm4 game engine and it has a similar module size requirement.
> 
> Grain does do tree shaking and dead code elimination, but the upcoming optimizations allow them to work far more effectively.
> 
> Hope that's helpful!

Hi thank you for your response! really excited for Grain v0.6 i really like the language, is there anything i can do temporarily to reduce the module size, can you guide me in any way ? otherwise its totally fine i was just curious thanks alot !
